builder/googlecompute/step_create_instance.go

```diff
@@ -24,13 +24,17 @@ func (c *Config) createInstanceMetadata(sourceImage *Image, sshPublicKey string)
 		instanceMetadata[k] = v
 	}
 
-	// Merge any existing ssh keys with our public key.
-	sshMetaKey := "sshKeys"
```

Comment:
Originally, packer would generate a private & public key for you, and attach that to the instance. With this change, if you do specify a private_key, packer won't modify your instance's ssh metadata, and will attempt to use your supplied key to authenticate.

The motivation for this change is that there are currently some GCE images that can't seem to read their account metadata, and instance-specific keys will fail to authenticate. However, they can make use of [project-wide](https://cloud.google.com/compute/docs/instances/adding-removing-ssh-keys#project-wide) ssh keys. The workflow now becomes:

  1. Generate your key pair.
  2. Upload the public key to GCE.
  3. Point packer at the private key.

Given packer's behavior when you don't specify a private_key, I can't seem to think of a use-case where you would prefer to attach an instance-specific key to an instance, rather than use a temporary key pair generated by packer.